### DISCUSSION TITLE
Integrated with LangChain?

### DISCUSSION THREAD

**Discussion by tap2k** [eyurtsev/kor#278] [General]
create_extraction_chain is now included as part of LangChain:
https://api.python.langchain.com/en/latest/chains/langchain.chains.openai_functions.extraction.create_extraction_chain.html

Does this mean the kor library is no longer needed?

**Comment by eyurtsev** [comment]
There are different approaches to extraction using LLMs: https://python.langchain.com/docs/use_cases/extraction/

Kor does a better job than most libraries for prompt based parsing. So if you're working with a model that does not support function calling, I'd recommend using Kor (or exploring JSON mode if that's supported by the model).

If you want to use LangChain function calling look through the guide I referenced above -- you should be using the with_structured_output -- and definitely include examples.


----

create_extraction_chain in langchain bears the same name but behaves differently.

**Comment by tap2k** [comment]
Ah, I see. Thanks for the clarification!
